ServerCore/Pages/Hints/AuthorIndex.cshtml.cs

```diff
@@ -0,0 +1,166 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Identity;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.EntityFrameworkCore;
+using ServerCore.DataModel;
+using ServerCore.Helpers;
+using ServerCore.ModelBases;
+
+namespace ServerCore.Pages.Hints
+{
+    [Authorize(Policy = "IsEventAdminOrEventAuthor")]
+    public class AuthorIndexModel : EventSpecificPageModel
+    {
+        public AuthorIndexModel(PuzzleServerContext serverContext, UserManager<IdentityUser> userManager) : base(serverContext, userManager)
+        {
+        }
+
+        public List<HintStatePerTeam> HintStatePerTeam { get; set; }
+
+        public Puzzle Puzzle { get; set; }
+
+        public Team  Team { get; set; }
+
+        public SortOrder? Sort { get; set; }
+
+        public const SortOrder DefaultSort = SortOrder.TimeDescending;
+
+        public async Task<IActionResult> OnGetAsync(int? puzzleId, int? teamId, SortOrder? sort)
+        {
+            Sort = sort;
+
+            if (puzzleId == null)
+            {
+                if (EventRole == EventRole.admin)
+                {
+                    if (teamId == null)
+                    {
+                        HintStatePerTeam = await _context.HintStatePerTeam.Where((h) => h.UnlockTime != null && h.Team.Event == Event).ToListAsync();
+                    }
+                    else
+                    {
+                        HintStatePerTeam = await _context.HintStatePerTeam.Where((h) => h.UnlockTime != null && h.TeamID == teamId).ToListAsync();
+                    }
+                }
+                else
+                {
+                    // Surely there is a way to get a join to do a bunch of this work, but joins are simply not for me. Someone else can fix later.
```

Comment:
We'll have to keep an eye on the efficiency of this. Without a join, it could get expensive.

Reply:
My feeling is that this is a rare page and the number of hints taken will generally not be overly large. I'm happy if someone else writes a join here, but I find the syntax too baffling to write it myself.